StringIndexOutOfBoundsException is thrown when computing getter methods

https://github.com/redhat-developer/quarkus-ls/blob/b16a72ca1a445e2e5256ce21289aa970e5cfacc8/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/commons/JavaMethodInfo.java#L197

When calling completion for a bean, JavaMethodInfo.computeGetterName() throws a StringIndexOutOfBoundsException if the class contains a method called `get()` (should also fail if it contains an `is()` method).

```
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: 3
	at java.base/java.lang.StringLatin1.charAt(Unknown Source)
	at java.base/java.lang.String.charAt(Unknown Source)
	at com.redhat.qute.commons.JavaMethodInfo.computeGetterName(JavaMethodInfo.java:197)
	at com.redhat.qute.commons.JavaMethodInfo.getGetterName(JavaMethodInfo.java:178)
	at com.redhat.qute.services.completions.QuteCompletionsForExpression.fillCompletionMethods(QuteCompletionsForExpression.java:341)
	at com.redhat.qute.services.completions.QuteCompletionsForExpression.doCompleteForJavaTypeMembers(QuteCompletionsForExpression.java:244)
	at com.redhat.qute.services.completions.QuteCompletionsForExpression.lambda$doCompleteExpressionForMemberPart$1(QuteCompletionsForExpression.java:211)
```

--- a/qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/commons/JavaMethodInfo.java
+++ b/qute.jdt/com.redhat.qute.jdt/src/main/java/com/redhat/qute/commons/JavaMethodInfo.java
@@ -27,6 +27,10 @@ import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
  */
 public class JavaMethodInfo extends JavaMemberInfo {
 
+	private static final String GET_PREFIX = "get";
+
+	private static final String IS_PREFIX = "is";
+
 	private transient String methodName;
 
 	private transient String returnType;
@@ -185,10 +189,13 @@ public class JavaMethodInfo extends JavaMemberInfo {
 			return NO_VALUE;
 		}
 		String methodName = getName();
+		if (GET_PREFIX.equals(methodName) || IS_PREFIX.equals(methodName)) {
+			return NO_VALUE;
+		}
 		int index = -1;
-		if (methodName.startsWith("get")) {
+		if (methodName.startsWith(GET_PREFIX)) {
 			index = 3;
-		} else if (methodName.startsWith("is")) {
+		} else if (methodName.startsWith(IS_PREFIX)) {
 			index = 2;
 		}
 		if (index == -1) {

--- a/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/commons/JavaMethodInfo.java
+++ b/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/commons/JavaMethodInfo.java
@@ -27,6 +27,10 @@ import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
  */
 public class JavaMethodInfo extends JavaMemberInfo {
 
+	private static final String GET_PREFIX = "get";
+
+	private static final String IS_PREFIX = "is";
+
 	private transient String methodName;
 
 	private transient String returnType;
@@ -185,10 +189,13 @@ public class JavaMethodInfo extends JavaMemberInfo {
 			return NO_VALUE;
 		}
 		String methodName = getName();
+		if (GET_PREFIX.equals(methodName) || IS_PREFIX.equals(methodName)) {
+			return NO_VALUE;
+		}
 		int index = -1;
-		if (methodName.startsWith("get")) {
+		if (methodName.startsWith(GET_PREFIX)) {
 			index = 3;
-		} else if (methodName.startsWith("is")) {
+		} else if (methodName.startsWith(IS_PREFIX)) {
 			index = 2;
 		}
 		if (index == -1) {

--- a/qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/commons/JavaMethodInfoTest.java
+++ b/qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/commons/JavaMethodInfoTest.java
@@ -12,6 +12,7 @@
 package com.redhat.qute.commons;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import org.junit.jupiter.api.Test;
@@ -62,5 +63,29 @@ public class JavaMethodInfoTest {
 
 		String resolved = method.resolveJavaElementType(listOfItem);
 		assertEquals("org.acme.Item", resolved);
+	}
+
+	@Test
+	public void getter() {
+		// getFoo()
+		JavaMethodInfo getFooMethod = new JavaMethodInfo();
+		getFooMethod.setSignature("getFoo() : int");
+		assertEquals("foo", getFooMethod.getGetterName());
+
+		// get()
+		JavaMethodInfo getMethod = new JavaMethodInfo();
+		getMethod.setSignature("get() : int");
+		assertNull(getMethod.getGetterName());
+
+		// isFoo()
+		JavaMethodInfo isFooMethod = new JavaMethodInfo();
+		isFooMethod.setSignature("isFoo() : boolean");
+		assertEquals("foo", isFooMethod.getGetterName());
+
+		// is()
+		JavaMethodInfo isMethod = new JavaMethodInfo();
+		isMethod.setSignature("is() : boolean");
+		assertNull(isMethod.getGetterName());
+
 	}
 }

